### PR TITLE
[Dashboard] Order results by favourites first by default

### PR DESCRIPTION
**Pull request by jesusbotella**
Please review only Frontend changes. The Backend code changes will go away whenever the backend branch I used to take this branch from is merged to `new-dashboard-feature` branch.

Related to #14372.

### CODE REVIEW

File: lib/assets/javascripts/new-dashboard/components/Settings/Ordering.vue [hunk 1]
```diff
@@ -2,6 +2,13 @@
 <div class="section">
   <h6 class="text is-xsmall is-txtSoftGrey u-tupper letter-spacing">{{ $t('SettingsDropdown.orderMaps') }}</h6>
   <ul class="list">
+    <li class="type text is-caption is-txtGrey" :class="{ 'type--selected': isOrderApplied('favorited,updated_at', 'desc,desc') }">
+      <a href="javascript:void(0)"
```
Comment (IagoLast) [inline lib/assets/javascripts/new-dashboard/components/Settings/Ordering.vue]: I think if you prevent default dont need the href void fix

File: lib/assets/javascripts/new-dashboard/store/datasets/index.js [hunk 2]
```diff
@@ -46,9 +52,9 @@ const datasets = {
 
     setURLOptions (state, options) {
       state.page = options.page ? parseInt(options.page) : 1;
-      state.filterType = options.filter || 'mine';
-      state.order = options.order || 'updated_at';
-      state.orderDirection = options.order_direction || 'desc';
+      state.filterType = options.filter || DEFAULT_VALUES.filter;
```
Comment (IagoLast) [inline lib/assets/javascripts/new-dashboard/store/datasets/index.js]: we can move this logic to multiple mutations as we did in maps

Comment (jesusbotella) [reply lib/assets/javascripts/new-dashboard/store/datasets/index.js]: I was about to do it for Your Data section in Home Page ;)

File: lib/assets/javascripts/new-dashboard/components/Settings/Ordering.vue (hunk 1 above)
Comment (jesusbotella) [reply lib/assets/javascripts/new-dashboard/components/Settings/Ordering.vue]: No, I left it there just for the pointer cursor ;) But I can set it via CSS and remove the `href` attribute.

**Comment by jesusbotella** [comment]
retest this please

**Pull request merged**